Sources/Shared/Data Manager/Model/Documents/File.m

```diff
@@ -74,8 +80,50 @@ - (void)dealloc {
     [_creator release];
     [_dateCreated release];
     [_dateModified release];
+    [natureName release];
     [super dealloc];
 }
 
+-(NSString *) getNatureName{
+    if (natureName) {
+        return natureName;
+    }
+    return _name;
+    
+}
+
+/*
+ In case of Group section, the name of the folder are genered automatically by the system & this names are not really nice (nature). The goal of this method is to convert this kind of name
+ */
+-(NSString* ) convertNameToNature {
+    
+    NSString * s = _name;
+    s = [s stringByReplacingOccurrencesOfString:@"." withString:@" "];
+    s = [s stringByReplacingOccurrencesOfString:@"_" withString:@" "];
+    s = [s stringByReplacingOccurrencesOfString:@"exo" withString:@" eXo"];
+    s = [s stringByReplacingOccurrencesOfString:@" spaces " withString:@" space "];
+    while ([s rangeOfString:@"  "].location !=NSNotFound) {
+        s = [s stringByReplacingOccurrencesOfString:@"  " withString:@" "];
+    }
+    
+    //First character of a word would be in Uppercase
+    for (int index =0; index < s.length -1 ; index++ ){
+        if ([s characterAtIndex:index] == ' ' ){
+            s = [s stringByReplacingCharactersInRange:NSMakeRange(index+1,1) withString:[[s substringWithRange:NSMakeRange(index+1,1)] uppercaseString]];
+        }
+    }
+    
+    s = [s stringByReplacingOccurrencesOfString:@"EXo" withString:@"eXo"];
+    while ([s characterAtIndex:0] == ' ') {
+        s = [s substringFromIndex:1];
+    }
+    while ([s characterAtIndex:(s.length-1)] == ' ') {
+        s = [s substringToIndex:s.length-1];
+    }
+    natureName = s;
```

Comment:
You should `release` natureName before affecting it a new value (if it's not nil). Otherwise the previous value is leaked.